mitosheet/src/mito/components/modals/UserEditedCodeModal.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mito
+
+import React, { Fragment } from 'react';
+import { MitoAPI } from '../../api/api';
+import { UIState, UserProfile } from '../../types';
+import DefaultModal from '../DefaultModal';
+import TextButton from '../elements/TextButton';
+import { ModalEnum } from './modals';
+
+/*
+    This modal displays to the user when:
+    1. the analysis that they are replaying does not exist on their computer
+    2. the analysis errors during replay for some other reason
+*/
+const UserEditedCodeModal = (
+    props: {
+        setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+        mitoAPI: MitoAPI,
+        userProfile: UserProfile,
+        onClickButton: (overwriteUserEdits: boolean) => void,
+    }): JSX.Element => {
+    return (
+        <DefaultModal
+            header='Insert New Cell?'
+            modalType={ModalEnum.Error}
+            wide
+            viewComponent={
+                <Fragment>
+                    <div className='text-align-left text-body-1'>
+                        Looks like you edited the code generated by Mito in the cell below. Do you want to overwrite these edits or insert a new cell for the Mito generated code?
+                    </div>
+                </Fragment>
+            }
+            buttons={
+                <>
+                    <TextButton 
+                        variant='light'
+                        width='hug-contents'
+                        onClick={() => {
+                            props.onClickButton(true);
```

Comment:
Why I think it's overabstracted: it's totally unclear what this function is doing when you read this component. I missed it on my first time through.

Reply:
On top of moving this function into this component, I think we should rename it to something a bit more communicative.